website/src/blogs/2018-11-27-experience-with-openebs-in-this-hacktoberfest.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Experience with OpenEBS in this Hacktoberfest
+author: Aswath
+author_info: FullStack Dev
+date: 26-11-2018
+tags: OpenEBS, Open Source, Hacktoberfest
+excerpt: From Hacktoberfest website, I came to know that Open Source Cafe Bangalore is conducting Hacktoberfest workshops every weekend and they are also providing some awesome goodies for our contributions. 
+---
+
+From Hacktoberfest website, I came to know that Open Source Cafe Bangalore is conducting Hacktoberfest workshops every weekend and they are also providing some awesome goodies for our contributions. So, I decided to give it a try.
+
+I was late for the workshop on day one itself (Thanks to traffic in Bangalore!). A session about `Kubernetes` was ongoing. Since I don’t even know what a `Container` is, it was hard for me to understand what `Kubernetes` is all about. The next session was about how we can make our first PR to some open source repository. Since I already made a few PRs, I understood it.
+
+## It’s better to start with simple issues first.
+
+In most of the repositories, people will tag the issues concerning the complexity. Thanks to the [openebs](https://medium.com/@openebs) community for tagging beginner friendly issues. So, I choose to work on a few issues tagged as `hacktoberfest` — a few of them were very simple, such as fixing the linting errors. But a few of them were little complicated like [adding autocompletion to a CLI tool](https://github.com/openebs/openebs/issues/1987).
+Taking up the more straightforward issues initially gave me confidence. And it helped me to understand the conventions to follow for that particular repository such as signing every commit that we make.
+
+## Help is available in the community if you need.
+
+When I decided to work on the project OpenEBS/Maya, I struggled to set up the project on my machine. I was using MacOS, and the build instructions were written for Linux OSes. People like [Satyam Zode](https://medium.com/@satyamz) and Prateek helped me to build the project by changing some lines in the `Makefile`.
+This project was written entirely in `Golang`. When I decided to work on this project, I didn’t even know how to write a hello world program in `Golang`. But people were there to help me whenever I got stuck.
+
+## Always consult with some maintainers before you add some external libraries to the project.
+
+I took the issue of adding bash completion to a CLI tool named mayactl, since Akshay has mentioned a library to do the same thing. I learned a bit of `Golang` from A Tour of Go. Then I tried to incorporate the bash completion library to mayactl. After a couple of failed attempt, I was able to implement the essential bash completion. In the review, [Ashutosh Kumar](https://medium.com/@sonasingh46) told me that if we add an external library to our project, we should also be able to maintain it. So, he asked me to decide on it.
```

Comment:
```suggestion
I took the issue of adding bash completion to a CLI tool named mayactl, since Akshay has mentioned a library to do the same thing. I learned a bit of `Golang` from A Tour of Go. Then I tried to incorporate the bash completion library to mayactl. After a couple of failed attempts, I was able to implement the essential bash completion. In the review, [Ashutosh Kumar](https://medium.com/@sonasingh46) told me that if we add an external library to our project, we should also be able to maintain it. So, he asked me to decide on it.
```